src/items.ts

```diff
@@ -1,124 +1,1518 @@
+/**
+ * items.ts
+ * All available items and their properties
+ * lgfrbcsgo & Nikolaus - October 2020
+ */
+
 export type Liter = number
 export type Minutes = number
 export type Quantity = number
 
-export type Item = Machine | Container | Ore | Other
+export type Item = Ore | Catalyst | Product | Element
 
 export enum ItemType {
-    MACHINE = "machine",
-    CONTAINER = "container",
     ORE = "ore",
-    OTHER = "product",
+    CATALYST = "catalyst",
+    PRODUCT = "product",
+    ELEMENT = "element",
 }
 
-export interface Container extends CommonProperties {
-    readonly type: ItemType.CONTAINER
+export enum Tier {
+    BASIC = "basic",
+    UNCOMMON = "uncommon",
+    ADVANCED = "advanced",
+    RARE = "rare",
+    EXOTIC = "exotic",
+    ELEMENT = "element",
 }
 
-export interface Machine extends CommonProperties {
-    readonly type: ItemType.MACHINE
+export enum Category {
+    AIRFOIL = "airfoil",
+    ANTI_GRAVITY = "anti-gravity",
+    CATALYST = "catalyst",
+    COMPLEX_PARTS = "complex parts",
+    CONTROL = "control",
+    CORE_UNIT = "core unit",
+    ENGINES = "engines",
+    EXCEPTIONAL_PARTS = "exceptional parts",
+    FLIGHT_CONTROL = "flight control",
+    FUEL = "fuel",
+    FUEL_TANKS = "fuel tanks",
+    FUNCTIONAL_PARTS = "functional parts",
+    INDUSTRY = "industry",
+    INSTRUMENTS = "instruments",
+    INTERACTIVE = "interactive",
+    INTERMEDIARY_PARTS = "intermediary parts",
+    ITEM_CONTAINERS = "item containers",
+    ORE = "ore",
+    PRODUCT = "product",
+    PURE = "pure",
+    SENSOR = "sensor",
+    STRUCTURAL_PARTS = "structural parts",
 }
 
-export interface Other extends CommonProperties {
-    readonly type: ItemType.OTHER
+/**
+ * Properties commen to each item type
+ */
+export interface CommonProperties {
+    readonly name: string
+    readonly tier: string
+    readonly category: string
```

Comment:
```suggestion
    readonly tier: Tier
    readonly category: Category
```

Reply:
Fixed in https://github.com/lgfrbcsgo/du-factory-generator/commit/9413cb40975215110ead9dad189d664c82f8b980